legate/core/legate.py

```diff
@@ -127,30 +128,30 @@ def __init__(self, schema: pyarrow.Schema, columns: list[Array]) -> None:
         self._columns = columns
 
     @property
-    def __legate_data_interface__(self) -> dict[str, Any]:
+    def __legate_data_interface__(self) -> LegateDataInterfaceItem:
         """
         The Legate data interface allows for different Legate libraries to get
         access to the base Legion primitives that back objects from different
         Legate libraries. It currently requires objects that implement it to
-        return a dictionary that contains two integer members:
+        return a dictionary that contains two members:
 
         Returns
         -------
         A dictionary with the following entries:
+
         'version' (required) : int
             An integer showing the version number of this implementation of
             the interface (i.e. 1 for this version)
-        'data' (required) : OrderedDict[Field,Array]
-            An ordered dictionary mapping 'Field' objects that represent the
-            names and types of the field data to 'Array' objects containing
+
+        'data' (required) : dict[Field, Array]
+            An dictionary mapping ``pyarrow.Field`` objects that represent the
+            names and types of the field data to ``Array`` objects containing
```

Comment:
`dict` is ordered for all versions of Python we support.

Reply:
@magnatelee @lightsighter FYI this changes the specification of the legate data interface, to align with all existing (technically wrong) uses of dict rather than OrderedDict.